main/src/library/File.flix

```diff
@@ -59,4 +159,180 @@ namespace File {
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath, forName(opts.charSet));
+            let fileRead = readHelper(readAll(reader));
+            close(reader);
+            Ok(readWithHelper(fileRead, opts.offset, opts.count))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+    }
+
+    ///
+    /// Helper function for readWith.
+    /// Takes a String and applys the options.
+    ///
+    def readWithHelper(f: String, offset: Int64, count: Int32): String & Pure =
+        let fileLength = String.length(f);
+        match Int64.tryToInt32(offset) {
+            case None => String.dropRight(fileLength - count, f)
+            case Some(n) =>
+                let str = String.dropLeft(n, f);
+                let strLength = String.length(str);
+                String.dropRight(strLength - count, str)
+        }
+
+    ///
+    /// Returns an iterator of the given file `f`
+    ///
+    pub def readLinesIter(f: String): Iterator[Result[String, String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path);
+            import java.io.BufferedReader.readLine();
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath);
+            let line = ref readLine(reader);
+
+            let done = () -> Object.isNull(deref line);
+            let next = () -> {
+                let l = deref line;
+                line := readLine(reader);
+                Ok(l)
+            };
+            Iterator(done, next)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Iterator.singleton(Err(getMessage(ex)))
+        }
+
+    ///
+    /// Returns an array of all the bytes in the given file `f`.
+    ///
+    pub def readBytes(f: String): Result[Array[Int8], String] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:readAllBytes(##java.nio.file.Path);
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let bytes = readAllBytes(javaPath);
+
+            Ok(bytes)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns an iterator of the bytes in the given `file` in chunks of size `chunkSize`.
+    ///
+    pub def readChunks(f: String, chunkSize: Int32): Iterator[Result[Array[Int8], String]] & Impure =
```

Comment:
If you want to use this method it is probably to avoid having the whole array in memory but the `bytes` variable contained in the iterator closure contains the whole file. The iterator should probably use [`read()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/BufferedReader.html#read()) (this int is actually a byte, I don't know if there's a method which actually returns the Byte type)